tests/Twig/Extension/SonataAdminExtensionTest.php

```diff
@@ -2408,10 +2494,14 @@ public function testIsGrantedAffirmative(): void
      */
     public function testRenderViewElementCompare(string $expected, string $type, $value, array $options, ?string $objectName = null): void
     {
-        $this->admin
+        $this->templateRegistry
             ->method('getTemplate')
             ->willReturn('@SonataAdmin/CRUD/base_show_compare.html.twig');
 
+        $this->admin
+            ->method('getTemplateRegistry')
+            ->willReturn($this->templateRegistry);
+
```

Comment:
Admin::getTemplate doesn't exist anymore.

Admin::getTemplateRegistry is a final protected method so I can't mock it.

How do I solve this ? @phansys @sonata-project/contributors ?

Reply:
You could use a real `TemplateRegistry` object. It should result in less mocking, which is better anyway IMO.